### ISSUE TITLE
Upgrading from 13.2.0 to 13.2.1 breaks ability to run Behat tags by specific tag (and possibly ability to override any BLT config?)

### ISSUE THREAD

**Issue by bkosborne** [acquia/blt#4456]
**Describe the bug**
When writing behat tests, I'll often tag them and run specific tags only using this command line:
```
blt tbr -D behat.tags=test123
```

After updating from 13.2.0 to 13.2.1, this doesn't work anymore, and it is running all my tests.

Here's the command that BLT generates in 13.2.0:
```
[Testing\Behat] Running behat  --format pretty /var/www/html/princeton/tests/behat --colors --no-interaction --stop-on-failure --strict --config /var/www/html/princeton/tests/behat/local.yml --profile default --tags test123 -v
```

And here's what it generates in 13.2.1:
```
[Testing\Behat] Running behat  --format pretty /var/www/html/princeton/tests/behat --colors --no-interaction --stop-on-failure --strict --config /var/www/html/princeton/tests/behat/local.yml --profile default --tags '~ajax&&~experimental&&~lightningextension' -v
```

So it's passing `~ajax&&~experimental&&~lightningextension` as the tags instead of the override I provided. These tags it's using are default in build.yml. I guess this upgrade somehow broke the ability to override BLT config values from the command line?

**To Reproduce**
See above

**Expected behavior**
Can run a specific tag in Behat instead of the whole suite by passing the tags to the BLT command line.

**Detailed error output**
N/A

**BLT doctor output**
```
+-----------------------+--------------------------------------------------------------+
| Property              | Value                                                        |
+-----------------------+--------------------------------------------------------------+
| %paths.%root          | /var/www/html/princeton/docroot                              |
| %paths.%site          | sites/default                                                |
| %paths.%modules       | sites/all/modules                                            |
| %paths.%themes        | sites/all/themes                                             |
| %paths.%config-sync   | /var/www/html/princeton/config/default                       |
| %paths.%files         | sites/default/files                                          |
| %paths.%temp          | /tmp                                                         |
| %paths.%private       | /var/www/html/princeton/files-private/default                |
| admin-theme           | ps_seven                                                     |
| alias-searchpaths.0   | /var/www/html/princeton/drush/sites                          |
| base-profile          | This profile does not extend a base profile.                 |
| blt-version           | 13.2.1.0                                                     |
| bootstrap             | Successful                                                   |
| composer-version      | Composer version 2.0.12 2021-04-01 10:14:59                  |
| config-sync           | /var/www/html/princeton/config/default                       |
| db-driver             | mysql                                                        |
| db-hostname           | db                                                           |
| db-name               | default_site                                                 |
| db-password           |                                                              |
| db-port               | 3306                                                         |
| db-status             | Connected                                                    |
| db-username           | root                                                         |
| drupal-settings-file  | sites/default/settings.php                                   |
| drupal-version        | 9.2.10                                                       |
| drush-cache-directory | /root/.drush/cache                                           |
| drush-conf.0          | /var/www/html/princeton/vendor/drush/drush/drush.yml         |
| drush-conf.1          | /var/www/html/princeton/drush/drush.yml                      |
| drush-conf.2          | /var/www/html/princeton/docroot/sites/default/local.drush.ym |
|                       | l                                                            |
| drush-script          | /var/www/html/princeton/vendor/drush/drush/drush             |
| drush-temp            | /tmp                                                         |
| drush-version         | 10.3.6                                                       |
| files                 | sites/default/files                                          |
| install-profile       | ps                                                           |
| modules               | sites/all/modules                                            |
| php-bin               | /usr/local/bin/php                                           |
| php-conf.1            | false                                                        |
| php-os                | Linux                                                        |
| private               | /var/www/html/princeton/files-private/default                |
| root                  | /var/www/html/princeton/docroot                              |
| site                  | sites/default                                                |
| temp                  | /tmp                                                         |
| theme                 | ps_tiger                                                     |
| themes                | sites/all/themes                                             |
| uri                   | https://princeton-sites.localhost.princeton.edu              |
+-----------------------+--------------------------------------------------------------+
```

**System information**
* Operating system type: Docker
* BLT version: 13.2.1

**Comment by danepowell** [comment]
If it's true that this is a regression in 13.2.1, it can only be caused by https://github.com/acquia/blt/pull/4433 . And it would only affect projects using League Container 3.x.

However, I cannot reproduce this using BLT 13.2.1 and League Container 3.x, the `-D` argument works fine.

Having said that, it does seem like there might be a bug just looking at the code, even though I can't reproduce it. Can you see if https://github.com/acquia/blt/pull/4457 helps?

**Comment by bkosborne** [comment]
Ugh, although now I get this error:

```
blt tbr -Dbehat.tags=test123

Fatal error: Uncaught Error: Call to undefined method League\Container\Definition\Definition::withArgument() in /var/www/html/princeton/vendor/acquia/blt-behat/src/Blt/Plugin/Commands/BehatTestCommand.php:31
Stack trace:
#0 /var/www/html/princeton/vendor/consolidation/annotated-command/src/Hooks/Dispatchers/InitializeHookDispatcher.php(47): Acquia\BltBehat\Blt\Plugin\Commands\BehatTestCommand->initialize(Object(Symfony\Component\Console\Input\ArgvInput), Object(Consolidation\AnnotatedCommand\AnnotationData))
#1 /var/www/html/princeton/vendor/consolidation/annotated-command/src/Hooks/Dispatchers/InitializeHookDispatcher.php(36): Consolidation\AnnotatedCommand\Hooks\Dispatchers\InitializeHookDispatcher->doInitializeHook(Array, Object(Symfony\Component\Console\Input\ArgvInput), Object(Consolidation\AnnotatedCommand\AnnotationData))
#2 /var/www/html/princeton/vendor/consolidation/annotated-command/src/Hooks/Dispatchers/InitializeHookDispatcher.php(29): Consolidation\AnnotatedCommand\Hooks\Dispatchers\InitializeHookDispatcher- in /var/www/html/princeton/vendor/acquia/blt-behat/src/Blt/Plugin/Commands/BehatTestCommand.php on line 31
```

**Comment by danepowell** [comment]
Ah okay, the fact that this only happens on the legacy container was actually opposite of my expectation. But now I can reproduce it by requiring `league/container:^2` and running something like `blt status -D drush.alias=foo` just to see if parameters are respected.

I'll fix this in #4457 and cut a new release immediately.
